### PR TITLE
fix(tx-fetcher): only remove peer from active_peers when inflight_count <= 0

### PR DESCRIPTION
**Pull request by int88**
fix logic error of `decrement_inflight_request_count_for`.

### CODE REVIEW

File: crates/net/network/src/transactions/fetcher.rs [hunk 1]
```diff
@@ -90,10 +90,10 @@ impl TransactionFetcher {
     fn decrement_inflight_request_count_for(&mut self, peer_id: &PeerId) {
         let remove = || -> bool {
             if let Some(inflight_count) = self.active_peers.get(peer_id) {
-                if *inflight_count <= DEFAULT_MAX_COUNT_CONCURRENT_REQUESTS_PER_PEER {
+                *inflight_count -= 1;
```
Comment (emhane) [inline crates/net/network/src/transactions/fetcher.rs:93]:
```suggestion
                *inflight_count.saturating_sub(1);
```

better safe than sorry

**Review by emhane — CHANGES_REQUESTED** [review]
nice find! this hasn't been a problem now, since `DEFAULT_MAX_COUNT_CONCURRENT_REQUESTS_PER_PEER` is 1. hence when a request resolves, the peer should indeed always be removed from active peers. **but**, the code isn't future proof for when we make the max count configurable.

please when you find a bug like this, add a test in the pr that fixes it to cover the buggy code.

thanks!

**Comment by int88** [comment]
> 
> please when you find a bug like this, add a test in the pr that fixes it to cover the buggy code.
> 
will do :)

File: crates/net/network/src/transactions/fetcher.rs (hunk 1 above)
Comment (rkrasiuk) [reply crates/net/network/src/transactions/fetcher.rs:93]: i'm a proponent of using checked methods in case of unintended behavior such as this, thoughts?

**Comment by int88** [comment]
@emhane updated :)

**Review by emhane — APPROVED** [review]
lgtm

**Pull request merged**